gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineTesting.java

```diff
@@ -61,38 +63,44 @@ public void apply(Project project) {
         });
 
         project.getPlugins().withType(JavaPlugin.class, unusedPlugin -> {
-            // afterEvaluate necessary because the junit-jupiter dep might be added further down the build.gradle
-            project.afterEvaluate(proj -> {
-                proj.getConvention()
-                        .getPlugin(JavaPluginConvention.class)
-                        .getSourceSets()
-                        .matching(ss -> hasCompileDependenciesMatching(proj, ss, BaselineTesting::isJunitJupiter))
-                        .forEach(ss -> {
-                            Optional<Test> maybeTestTask = getTestTaskForSourceSet(proj, ss);
-                            if (!maybeTestTask.isPresent()) {
-                                log.warn("Detected 'org:junit.jupiter:junit-jupiter', but unable to find test task");
-                                return;
-                            }
-                            log.info(
-                                    "Detected 'org:junit.jupiter:junit-jupiter', enabling useJUnitPlatform() on {}",
-                                    maybeTestTask.get().getName());
-                            enableJunit5ForTestTask(maybeTestTask.get());
-
-                            // Also wire up a test ignore for this source set
-                            project.getPlugins().withType(BaselineExactDependencies.class, exactDeps -> {
-                                TaskContainer tasks = project.getTasks();
-                                tasks.named(
-                                        BaselineExactDependencies.checkUnusedDependenciesNameForSourceSet(ss),
-                                        CheckUnusedDependenciesTask.class,
-                                        task -> task.ignore("org.junit.jupiter", "junit-jupiter"));
-                            });
+            TaskProvider<CheckJUnitDependencies> checkJUnitDependencies =
+                    project.getTasks().register("checkJUnitDependencies", CheckJUnitDependencies.class);
+
+            project.getConvention()
+                    .getPlugin(JavaPluginConvention.class)
```

Comment:
I think we should use the `JavaPluginExtension` as conventions are deprecated and to be removed with Gradle 8.

Reply:
I tried, but Gradle 6 doesn't have the `getSourceSets` method we need https://github.com/palantir/gradle-baseline/pull/1929/commits/118cb972f812a11e3b781f90e29064c10efc4439

Reply:
Ah I remember encountering this before. I guess Gradle 7 is the bridge version then at with Gradle 8 we have to cut support for Gradle 6. Fine for now then!